core/controllers/profile.py

```diff
@@ -139,6 +153,24 @@ def post(self, _):
 class PreferencesHandler(base.BaseHandler):
     """Provides data for the preferences page."""
 
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'update_type': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
```

Comment:
If you look at the code we compare `update_type` with some predefined values, so you can probably restrict available values for the `update_type`.

Reply:
I have taken `choices` type to restrict those values. please me let me know if any other changes needed.